### PR TITLE
Authentication added to cart and wishlist routes.

### PR DESCRIPTION
**Pull request by Manju-sree**
Authentication feature added to ecommerce App.
- Access given to Cart and WishList Page only if user is loggedIn successfully to site .
-Login As Guest is implemented.
 You can also create your own login credentials using SignUp page and Login again with same email and password.

### CODE REVIEW

File: src/context/Reducer/AuthReducer.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+const AuthReducer = (authState, authAction) => {
+    const { type, payload } = authAction;
+    switch (type) {
+        case "LOG-IN":
+            return { ...authState, ...payload.foundUser, authenticationToken: payload.encodedToken, loginStatus: true }
+        case "LOG-OUT":
+            return { ...initialAuthState }
+        case "SIGN-UP":
+            return { ...authState, ...payload.createdUser, authenticationToken: payload.encodedToken, loginStatus: true }
```
Comment (JantuDeb) [inline src/context/Reducer/AuthReducer.jsx:14]: For login and signup you can use same return
```
case "LOG-IN":
case "SIGN-UP":
       return { ...authState, ...payload.user, authenticationToken: payload.encodedToken, loginStatus: true }
```

File: src/pages/Authentication/Login.jsx [hunk 2]
```diff
@@ -1,39 +1,109 @@
...
+    setLoginData((prevLoginData) => ({ ...prevLoginData, [name]: value }));
+  };
+  return (
+    <div className="login-container wd-100-pc pos-rel flex-grow display-flex-column">
+      <h1 className="auth-form-title">Login</h1>
+      <form
+        action="auth-form"
+        className="wd-100-pc display-flex-column"
+        onSubmit={(e) => {
+          LoginActionHandler(e, loginData, authDispatch, navigate);
```
Comment (JantuDeb) [inline src/pages/Authentication/Login.jsx:29]: No need to pass `e` to the function, you can do like this
```
onSubmit={(e) => {
          e.preventDefault()
          LoginActionHandler( loginData, authDispatch, navigate);

```

File: src/routes/ProtectedRoutes.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { useAuthentication } from "../context";
+import React from "react";
+import { Navigate } from "react-router-dom";
+
+export const ProtectedRoutes = ({ authToPages }) => {
+    const { authState: { loginStatus } } = useAuthentication();
+    return (
+        <>
+            {loginStatus ? authToPages : <Navigate replace to="/Login" />}
```
Comment (JantuDeb) [inline src/routes/ProtectedRoutes.jsx:9]: you can directly return this

**Review by JantuDeb — APPROVED** [review]
Overall the codes looks fine, go through the comments I added

File: src/pages/Authentication/Login.jsx (hunk 2 above)
Comment (Manju-sree) [reply src/pages/Authentication/Login.jsx:29]: Thanks for all sugesstions.

**Review by mariyasada — APPROVED** [review]
I didn't get some authentication code but the code structure is good. this pr is good to merge.

**Pull request merged**